src/NJsonSchema.CodeGeneration.CSharp/Models/PropertyModel.cs

```diff
@@ -33,8 +33,17 @@ public class PropertyModel : PropertyModelBase
             _property = property;
             _settings = settings;
             _resolver = typeResolver;
+            
+            Id = property.Id;
```

Comment:
Id = property.Id ?? property.ActualSchema.Id;

Can a property have its own id?
Maybe also use a derived property like the Name?